src/wallet/test/wallet_tests.cpp

```diff
@@ -65,6 +65,26 @@ static bool equal_sets(CoinSet a, CoinSet b)
     return ret.first == a.end() && ret.second == b.end();
 }
 
+BOOST_AUTO_TEST_CASE(second_loop_test)
+{
+    CoinSet setCoinsRet;
+    CAmount nValueRet;
+    LOCK(wallet.cs_wallet);
+ 
+    empty_wallet();
+    add_coin(2 * CENT);
+    for (int i = 0; i < 1000; i++)
+        add_coin(3 * CENT);
+    for (int i = 0; i < 1000; i++)
+        add_coin(1000 * CENT);
+
+    BOOST_CHECK(wallet.SelectCoinsMinConf(1000001 * CENT, 1, 6, vCoins, setCoinsRet, nValueRet));
```

Comment:
@Xekyo You can either get the test case from #7293 and include it here or remove the test case in this pull.